### PR TITLE
Use schema v2

### LINKED ISSUE #7: Refactor place hierarchy from fixed tree to DAG
The current model (based on simpler assumptions of the scope of the project) was a fixed taxonomy of Country, State, and County. This works fairly well for the United States but does not scale to accommodate more complex relationships:

Countries like the United Kingdom and Sweden where this taxonomy doesn't match the regional breakdown. Cities that have multiple zip codes (multiple parents in the hierarchy). Cities like San Francisco that are both a city and a county. We should refactor the data, API, and UI to accommodate a more dynamic hierarchy of places that allows n-levels of nesting, multiple parents, and places to be categorized as different types.

Reference:
 - [API Issue](https://github.com/yahoo/covid-19-api/issues/6)

### PR DESCRIPTION
**Pull request by jkusa**
Resolves #7 

- Update queries and models based on schema v2
- Use location `id` instead of `wikiId` for lookups
- Remove "num" metrics
- Use and update search service
- Update logo
- Add 30px padding on left panel
- Import ember-fetch
- Update map settings to support more placeTypes
- Remove inflector util
- Remove location util
- Add ember-composable-helpers
- Support mirage in list filter escaped values
- Support mirage relationship field filters

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/case-details.js [hunk 1]
```diff
@@ -18,22 +18,16 @@ export default class CaseDetailsComponent extends Component {
...
-      totalDeaths = 0,
-      totalRecoveredCases = 0,
-      totalConfirmedCases = 0,
-    } = this.summaryRecord;
-    const unknown = totalConfirmedCases - (numActiveCases + totalDeaths + totalRecoveredCases);
+    const { totalDeaths = 0, totalRecoveredCases = 0, totalConfirmedCases = 0 } = this.summaryRecord;
+    const unknown = totalConfirmedCases - (totalDeaths + totalRecoveredCases);
     return [
-      { name: 'Active', data: [numActiveCases] },
       { name: 'Fatal', data: [totalDeaths] },
```
Comment (kevinhinterlong) [inline app/components/case-details.js:31]: Could intl these

File: app/components/map.js [hunk 2]
```diff
@@ -3,163 +3,192 @@
...
+      this.fetchLocationRecordsTask.perform();
     }
   }
 
-  @task(function* (location) {
-    if (['states', 'counties'].includes(location.type)) {
+  @task(function* () {
+    const { locationPlaceType } = this;
+    if (![SUPERNAME_TYPE, COUNTRY_TYPE].includes(locationPlaceType)) {
       const data = yield this.elide.fetch.perform('latestHealthRecords', {
```
Comment (kevinhinterlong) [inline app/components/map.js]:
```suggestion
      const latestHealthRecords = yield this.elide.fetch.perform('latestHealthRecords', {
```

File: app/components/map.js [hunk 3]
```diff
@@ -3,163 +3,192 @@
...
+        eq: {
+          'place.parents.parents.id': GLOBAL_ID,
         },
-        isNull: ['county', 'state'],
-        notNull: ['country'],
+        include: ['place'],
       }) || {};
 
-    const { states: stateRecords, countries: countryRecords } = yield hash({ states, countries });
+    const { stateAdminAreas: stateAdminAreaRecords, countries: countryRecords } = yield hash({
```
Comment (kevinhinterlong) [inline app/components/map.js]: We could initialize these as `*Records` and not do this renaming

File: app/services/location.js [hunk 4]
```diff
@@ -4,56 +4,46 @@
...
-        ...selectedLocation,
-        country,
-        state,
-        type,
-      };
-    } else {
-      return undefined;
-    }
+    const location = place?.data?.[0];
+    return location ? location : undefined;
```
Comment (kevinhinterlong) [inline app/services/location.js]: ?
```suggestion
    return place?.data?.[0];
```

**Review by kevinhinterlong — APPROVED** [review]
Small stuff, overall looks good

File: app/components/case-details.js [hunk 5]
```diff
@@ -18,22 +18,16 @@ export default class CaseDetailsComponent extends Component {
...
       return [];
     }
-    const {
-      numActiveCases = 0,
-      totalDeaths = 0,
-      totalRecoveredCases = 0,
-      totalConfirmedCases = 0,
-    } = this.summaryRecord;
-    const unknown = totalConfirmedCases - (numActiveCases + totalDeaths + totalRecoveredCases);
+    const { totalDeaths = 0, totalRecoveredCases = 0, totalConfirmedCases = 0 } = this.summaryRecord;
```
Comment (balajimanoharan) [inline app/components/case-details.js:28]: We aren't fetching `totalRecoveredCases` anymore. Can we clean this up?

File: app/components/distribution-chart.js [hunk 6]
```diff
@@ -21,7 +21,7 @@ export default class DistributionChartComponent extends Component {
           barHeight: '50%',
         },
       },
-      colors: ['#19c6f4', '#fed800', '#87d812', '#f2f2f2'],
+      colors: ['#fed800', '#87d812', '#f2f2f2'],
```
Comment (balajimanoharan) [inline app/components/distribution-chart.js:24]: Only need two colors? `Fatal` and `unknown`?

File: app/components/location-table.js [hunk 7]
```diff
@@ -7,110 +7,41 @@ import { action } from '@ember/object';
...
-      return location.country.attributes.wikiId;
-    }
-
-    return undefined;
-  }
-
-  get selectedLocationId() {
-    return this.args.location.attributes.wikiId;
+    const parent = location?.relationships.parents.data?.[0]?.id;
+    return parent;
```
Comment (balajimanoharan) [inline app/components/location-table.js]:
```suggestion
    return location?.relationships.parents.data?.[0]?.id;
```

File: app/controllers/dashboard/location.js [hunk 8]
```diff
@@ -16,11 +17,14 @@ export default class DashboardLocationController extends Controller {
   @service router;
 
   @tracked currentLocation;
+  @tracked locationLineage;
   @tracked mobileView = 'details';
 
+  GLOBAL_ID = GLOBAL_ID;
```
Comment (balajimanoharan) [inline app/controllers/dashboard/location.js]:
```suggestion
  globalId = GLOBAL_ID;
```

File: app/templates/dashboard/location.hbs [hunk 9]
```diff
@@ -4,10 +4,10 @@
   {{#unless this.screen.isMobile}}
     <aside class="dashboard__left">
       <CaseDetails
-        class="dashboard__global-case-details p-20"
+        class="dashboard__global-case-details p-x-30 p-y-20"
         @title={{t "globalConfirmedCases"}}
         @publishedDate={{this.metadata.publishedDate}}
-        @location={{hash attributes=(hash wikiId="Earth")}}
+        @location={{hash id=this.GLOBAL_ID}}
```
Comment (balajimanoharan) [inline app/templates/dashboard/location.hbs]:
```suggestion
        @location={{hash id=this.globalId}}
```

**Review by balajimanoharan — APPROVED** [review]
Small things 👍

File: app/components/case-details.js (hunk 1 above)
Comment (jkusa) [reply app/components/case-details.js:31]: In this case we don't show these as tooltips

File: app/components/case-details.js (hunk 5 above)
Comment (jkusa) [reply app/components/case-details.js:28]: Good catch, we should be fetching that field.

File: app/components/distribution-chart.js (hunk 6 above)
Comment (jkusa) [reply app/components/distribution-chart.js:24]: `Fatal`, `Recovered`, & `Unknown`

**Review by balajimanoharan — APPROVED** [review]

**Pull request merged**